ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +306,582 @@ def test_ipa_user_s4u2self_pac(self):
+        value
+        """
+        sec_rid_base_value = "103.2"
+        msg = (
+            "ipa-adtrust-install: error: option "
+            "--secondary-rid-base: invalid integer value: '{}'"
+        ).format(sec_rid_base_value)
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-A",
+                "admin",
+                "-a",
+                self.master.config.admin_password,
+                "--secondary-rid-base",
+                sec_rid_base_value,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stderr_text
+        assert result.returncode != 0
+
+    def test_adtrust_reinstall_updates_ipaNTFlatName_attribute(self):
+        """
+        Test checks that reinstalling ipa-adtrust-install with
+        new netbios name reflects changes in ipaNTFlatName attribute
+        and ipa trustconfig-show also reflects the same.
+        """
+        netbios_name = "TEST8REALM"
+        cmd = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-a",
+                self.master.config.admin_password,
+                "--netbios-name",
+                netbios_name,
+                "-U",
+            ]
+        )
+        assert cmd.returncode == 0
+        cmd_args = ["ldapsearch", "-Y", "GSSAPI", "(ipaNTFlatName=*)",
+                    "-s", "base", "-b", "cn="+"self.master.domain.name",
+                    "cn=ad,cn=etc,"+"self.master.domain.basedn"]
+        cmd = self.master.run_command(cmd_args)
+        cmd1 = self.master.run_command(["ipa", "trustconfig-show"])
+        assert "ipaNTFlatName: {}".format(netbios_name) in cmd.stdout_text
+        assert "NetBIOS name: {}".format(netbios_name) in cmd1.stdout_text
+
+    def test_smb_not_starting_post_adtrust_install(self):
+        """
+        Test checks that winbindd crash doesn't occur
+        and smb service is running post ipa-adtrust-install.
+        https://bugzilla.redhat.com/show_bug.cgi?id=991251
+        """
+        samba_msg = (
+            'Unit smb.service entered failed state'
+        )
+        core_dump_msg = (
+            'dumping core in /var/log/samba/cores/winbindd'
+        )
+        smb_cmd = self.master.run_command(
+            ['systemctl', 'status', 'smb']
+        )
+        assert smb_cmd.returncode == 0
+        assert samba_msg not in smb_cmd.stdout_text
+        winbind_cmd = self.master.run_command(
+            ['systemctl', 'status', 'winbind']
+        )
+        assert winbind_cmd.returncode == 0
+        assert core_dump_msg not in winbind_cmd.stdout_text
+
+    def test_samba_credential_cache_is_removed_post_uninstall(self):
+        """
+        Test checks that samba credential cache is removed after
+        ipa-server is uninstalled.
+        https://pagure.io/freeipa/issue/3479
+        """
+        result = tasks.install_master(self.master, setup_dns=True)
```

Comment:
Remove the call to install_master and ipa-adtrust-install, they were already done earlier.